R/visualization-functions.R

```diff
@@ -654,69 +654,192 @@ boxplot_vars <- function(md, include_vars, x_var) {
+          sex_outliers <- eval(parse( text = paste0( 'row.names( clean_metadata[ clean_metadata$', sex_var, '_Predicted != clean_metadata$', sex_var, ',])' ) ))
+          
+        }else{
+          #cluster 1 are females
+          fit$cluster[ fit$cluster == 1 ] <- 'female'
+          fit$cluster[ fit$cluster == 2 ] <- 'male'
+          eval(parse( text = paste0( 'clean_metadata$', sex_var, '_Predicted <-', NA) ))
+          eval(parse( text = paste0( 'clean_metadata$', sex_var, '_Predicted <- as.factor( as.character( fit$cluster[ row.names(clean_metadata) ] ) )' )))
+          sex_outliers <- eval(parse( text = paste0( 'row.names( clean_metadata[ clean_metadata$', sex_var, '_Predicted != clean_metadata$', sex_var, ',])' ) ))
+        }
+        
+        #Plot the data
+        ##Assemble the data
+        clean_metadata$discordant_by_sex <- 'NO'
+        clean_metadata[sex_outliers,]$discordant_by_sex <- 'YES'
+        voom_sexcounts <- limma::voom(sex_counts)  
+        results <- data.frame( SampleID = rownames(sex_pc$rotation),
+                                A = sex_pc$rotation[,xist_comp],
+                                B = sex_pc$rotation[,plot_component],
+                                XIST = as.numeric( voom_sexcounts$E[ row.names(biomart_results[ biomart_results$hgnc_symbol %in% c('XIST') , ]), rownames(sex_pc$rotation) ]),
+                                UTY = as.numeric( voom_sexcounts$E[ row.names(biomart_results[ biomart_results$hgnc_symbol %in% c('UTY') , ]), rownames(sex_pc$rotation) ]),
+                                indicated_sex = clean_metadata[ rownames(sex_pc$rotation), paste0( sex_var, '_Predicted') ],
+                                Sex_Predicted = clean_metadata[ rownames(sex_pc$rotation), paste0( sex_var, '_Predicted') ],
+                                discordant_by_sex = clean_metadata[ rownames(sex_pc$rotation), 'discordant_by_sex' ]
         )
-      )
-  results <- tidyr::pivot_wider(
-    dplyr::select(results, -chromosome_name, -geneId),
-    names_from = "hgnc_symbol",
-    values_from = "counts(log)"
-    )
-  results <- dplyr::left_join(results, md, "sampleId")
-
-  p <- ggplot2::ggplot(results, ggplot2::aes(x = .data$XIST, y = .data$UTY)) +
-    ggplot2::geom_point(ggplot2::aes(color = .data[[sex_var]])) +
-    sagethemes::scale_color_sage_d() +
-    sagethemes::theme_sage()
-  p <- list(plot = p,
-            sex_specific_counts = results)
-  p
+        
+        colnames(results)[ colnames(results) == 'A' ] <- paste0( 'PC', xist_comp )
+        colnames(results)[ colnames(results) == 'B' ] <-paste0( 'PC', plot_component )
+        
+        eigen <- sex_pc$sdev^2
+        pc_Comp <- signif( 100*(eigen[plot_component]/sum(eigen)), 3 )
+        pc_Sex <- signif( 100*(eigen[xist_comp]/sum(eigen)), 3)
+        
+        plot_markers <- ggplot2::ggplot( data = results  ) +
+          ggplot2::geom_point( ggplot2::aes( x=XIST, y=UTY, shape = indicated_sex, color = discordant_by_sex)  ) + 
```

Comment:
Look at line 703 for the syntax needed to call `ggplot` in a package and not throw errors.